static/app/views/dashboards/widgetCard/spansWidgetQueries.spec.tsx

```diff
@@ -206,4 +206,112 @@ describe('spansWidgetQueries', () => {
       })
     );
   });
+
+  it('triggers a preflight and then a best effort request for tables', async () => {
+    widget = WidgetFixture({
+      queries: [
+        {
+          name: '',
+          aggregates: ['a'],
+          fields: ['a'],
+          columns: [],
+          conditions: '',
+          orderby: '',
+        },
+      ],
+      displayType: DisplayType.TABLE,
+    });
+    const eventsStatsMock = MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/events/',
+      body: {
+        data: [{a: 'rendered'}, {a: 2}, {a: 3}],
+      },
+    });
+
+    const {rerender} = render(
+      <OrganizationContext.Provider
+        value={OrganizationFixture({
+          features: ['visibility-explore-progressive-loading'],
+        })}
+      >
+        <SpansWidgetQueries
+          api={api}
+          widget={widget}
+          selection={{
+            ...selection,
+            datetime: {period: '24hr', end: null, start: null, utc: null},
+          }}
+          dashboardFilters={{}}
+        >
+          {({tableResults}) => <div>{tableResults?.[0]?.data?.[0]?.a}</div>}
+        </SpansWidgetQueries>
+      </OrganizationContext.Provider>
+    );
+
+    expect(await screen.findByText('rendered')).toBeInTheDocument();
+    expect(eventsStatsMock).toHaveBeenCalledTimes(2);
+    expect(eventsStatsMock).toHaveBeenNthCalledWith(
+      1,
+      '/organizations/org-slug/events/',
+      expect.objectContaining({
+        query: expect.objectContaining({
+          sampling: 'PREFLIGHT',
+        }),
+      })
+    );
+    expect(eventsStatsMock).toHaveBeenNthCalledWith(
+      2,
+      '/organizations/org-slug/events/',
+      expect.objectContaining({
+        query: expect.objectContaining({
+          sampling: 'BEST_EFFORT',
+        }),
+      })
+    );
+
+    // Reset the mock so we can test that the rerender only triggers two requests
+    eventsStatsMock.mockReset();
+
+    // Rerender the component and check that the preflight is called before the best effort
+    rerender(
+      <OrganizationContext.Provider
+        value={OrganizationFixture({
+          features: ['visibility-explore-progressive-loading'],
+        })}
+      >
+        <SpansWidgetQueries
+          api={api}
+          widget={widget}
+          selection={{
+            ...selection,
+            datetime: {period: '1hr', end: null, start: null, utc: null},
+          }}
+          dashboardFilters={{}}
+        >
+          {({tableResults}) => <div>{tableResults?.[0]?.data?.[0]?.a}</div>}
+        </SpansWidgetQueries>
+      </OrganizationContext.Provider>
+    );
+
+    expect(await screen.findByText('rendered')).toBeInTheDocument();
```

Comment:
Not sure if it can be done here, but I usually try to avoid find by test and rely on `findByRole` if possible

Reply:
Since I'm mocking the child to just pull out the data I expect to return and putting it in a div, there isn't really any other UI/text that gets shown for now, but agreed, for normal components I'd prefer `ByRole`